File: server/src/Models/DMToolkitModels/Instances/FeatInstance.cs
using DMToolkit.API.Models.DMToolkitModels.Definitions;
using DMToolkit.API.Models.DMToolkitModels.Entities;

namespace DMToolkit.API.Models.DMToolkitModels.Instances;

public class FeatInstance : IDefinitionInstance<FeatDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public ICollection<Effect> FeatEffects { get; set; } = new List<Effect>();
    public string DefinitionId { get; set; } = string.Empty;
    public FeatDefinition? Definition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/featInstance.ts
import type { FeatDefinition } from "../definitions/featDefinition";
import type { Effect } from "../entities/effect";

export interface FeatInstance {
    id: string;
    effects: Effect[];
    definition: FeatDefinition;
}

export interface FeatInstanceDto {
    id: string;
    effectIds: string[];
    definitionId: string;
}